### ISSUE TITLE
`project_root` incorrectly resolved to parent repository when using git worktree inside the project directory

### ISSUE THREAD

**Issue by alpaca-tc** [rubocop/rubocop#15108]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

Other

### Steps to reproduce the problem

When running RuboCop inside a git worktree located under the parent repository (e.g. `<repo>/.worktrees/<name>`), `ConfigFinder.find_project_root` resolves to the parent repository's root instead of the worktree's root. This causes `.rubocop.yml` from both the parent and the worktree to be loaded, leading to duplicate `require` and `warning: already initialized constant` errors. 
https://github.com/rubocop/rubocop/blob/4e0d642eca6e9a694b8a359d39e0d4b5b6b26bb8/lib/rubocop/config_finder.rb#L34

```bash
mkdir -p project && cd project
git init

cat > Gemfile <<'RB'
source 'https://rubygems.org'
gem "rubocop"
RB
bundle install

cat > .rubocop.yml <<'YAML'
require:
  - ./lib/custom_cop.rb
YAML

mkdir -p lib
cat > lib/custom_cop.rb <<'RB'
class CustomCop < RuboCop::Cop::Base
  MSG = "something".freeze
end
RB

git add -A && git commit -m 'initial commit'

git worktree add .worktrees/test
cd .worktrees/test
bundle exec rubocop
```

### Expected behavior

RuboCop should recognize the worktree directory (`project/.worktrees/test/`) as the project root since it has its own `Gemfile` and `.rubocop.yml`. Only the worktree's `.rubocop.yml` should be loaded.

### Actual behavior

`project_root` resolves to `project/` (the parent repository). The parent's `.rubocop.yml` is also loaded, causing:

```
.../lib/custom_cop.rb:2: warning: already initialized constant CustomCop::MSG
.../.worktrees/test/lib/custom_cop.rb:2: warning: previous definition of MSG was here
```

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
1.86.1
```

---

## Workaround

Explicitly set `project_root` before other config is loaded:

```ruby
# lib/rubocop_worktree_project_root.rb
RuboCop::ConfigFinder.project_root = File.expand_path(File.join(__dir__, ".."))
```

```yaml
# .rubocop.yml
require:
  - ./lib/rubocop_worktree_project_root.rb
```

**Comment by alpaca-tc** [comment]
duplicated https://github.com/rubocop/rubocop/issues/11011
